### PR TITLE
Rewrite MessageList store logic and refactor chat store

### LINKED ISSUE #712: Chat updates (such as delete or new incomming message) result in jumping to the end
- Expected behavior: Scroll position is kept after new messages / deleting an old message
- Actual behavior: chat jumps to the end in these cases
- Steps to reproduce the problem: scroll a bit up and delete one of the earlier messages. 

This problem could be solved in the planned refactoring of the chatview #536

### PR DESCRIPTION
**Pull request by Jikstra**
- Moves fetch/paging logic into frontend and store
- Moves message convertion into backend
- fixes scroll jumping by scrolling to a specific message

Missing:
- [x] day markers
- [x] sending messages
- [x] deleting messages
- [x] update on incoming msgs
- [x] some message methods aren't working properly
- [x] iron all those bugs out
  - [x] `Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at module.exports../src/renderer/stores/MessageList.js.MessageListStore.effects.push (MessageList.js:123)`
  - [x] UI hangs on showing more message details
  - [x] restore default state when deleting currently selected chat
  - [x]  selecting a chat with unread messages needs two clicks
  - [x] sending a message doesn't scroll down to the message
- [x] make sure images have a fixed height

Fixes https://github.com/deltachat/deltachat-desktop/issues/1166
Fixes https://github.com/deltachat/deltachat-desktop/issues/1158
Fixes https://github.com/deltachat/deltachat-desktop/issues/477
Fixes https://github.com/deltachat/deltachat-desktop/issues/1180
Fixes https://github.com/deltachat/deltachat-desktop/issues/1122#issuecomment-561795867
Fixes #712

### CODE REVIEW

File: scss/message/_message.scss [hunk 1]
```diff
@@ -142,10 +142,10 @@
     }
 
     .module-message__img-attachment {
```
Comment (Simon-Laux) [inline scss/message/_message.scss]: some wide pictures get cut off, maybe make the height smaller in this case if possible.

File: src/renderer/stores/chat.js [hunk 2]
```diff
@@ -15,85 +16,215 @@ const defaultState = {
...
-      return msg.id === messageObj.id ? messageObj : msg
+chatStore.effects.push(async ({ type, payload }, state) => {
+  if (type === 'SELECT_CHAT') {
+    const chatId = payload
+    const chat = await callDcMethodAsync('chatList.selectChat', [chatId])
+    const messageIds = await callDcMethodAsync('messageList.getMessageIds', [chatId])
+    const oldestFetchedMessageIndex = messageIds.length - PAGE_SIZE
+    const newestFetchedMessageIndex = messageIds.length
+    const messageIdsToFetch = messageIds.slice(oldestFetchedMessageIndex, newestFetchedMessageIndex)
+    const messages = await callDcMethodAsync('messageList.getMessages', [messageIdsToFetch])
```
Comment (nicodh) [inline src/renderer/stores/chat.js]: Why do we need 3 times to call callDcMethodAsync?
Why is messageIDs not fetched together with chatList.selectChat
Why do we need to calculate messageIdsToFetch here? PAGE_SIZE could be a shared variable, so the calculation could be done also together with chatList.selectChat

File: scss/message/_message.scss (hunk 1 above)
Comment (Jikstra) [reply scss/message/_message.scss]: We can't dynamically set the height of images because this would make the scrollbar jump.

Comment (Simon-Laux) [reply scss/message/_message.scss]: I thought the saving image dimensions in the db api was already done, but now I see that its only partly implemented

**Pull request merged**